### ISSUE TITLE
ERROR:  Can't initialize DAQ pfring (-1)  - Fatal Error

### ISSUE THREAD

**Issue by dknobloc** [ntop/PF_RING#577]
While following your guide at https://www.ntop.org/guides/pf_ring/thirdparty/snort-daq.html, I have run into an error at step 3.5. Running snort in IDS mode.  The error message is:  "ERROR:  Can't initialize DAQ pfring (-1)  - Fatal Error, Quitting.."  My OS is RHEL Server Release 7.8 (3.120.0-1127-el7.x86_64).  I installed libpcap-1.8.1,  daq-2.0.6, openssl-1.1.1d, and snort-2.9.15.1 by compiling with no errors.  I can run "snort -v" with no errors, and "snort -T -c /etc/snort/snort.conf" with no errors.  I contacted ntop support, and it was suggested that I run pfcount on the same interface, and submit a github issue.  I then compiled pfcount using https://www.ntop.org/guides/pf_ring/examples.html as a guide.  I then ran "./pfcount -i eno2" and received the output "pf_ring_open error  [No such device] (pf_ring not loaded or interface eno2 is down."  When I do an "ifconfig eno2" a couple of times, I can verify that packets are being received as the packet count is going up.

**Comment by cardigliano** [comment]
Did you follow the instructions at https://www.ntop.org/guides/pf_ring/get_started/packages_installation.html to run the service (which loads the kernel module)?
Please provide: cat /proc/net/pf_ring/info

**Comment by dknobloc** [comment]
@cardigliano thank you for the suggestion.  The results of the cat was "No such file or directory."  I then followed the instructions in section 2.1 Configuration Wizard beginning with step 2.  Step 3 (pf_ringcfg --configure-driver igb --rss-queues 1) hung when installing igb-zc-5.3.5.18.1926-dkms.noarch.

**Comment by dknobloc** [comment]
Here is some more information regarding what I did before and after @cardigliano provided the suggestion to follow the instructions at https://www.ntop.org/guides/pf_ring/get_started/packages_installation.html.  I apologize if some of this information is not relevant.  

1. Compiled and installed libpcap-1.8.1 from source code.
2. Compiled and installed daq-2.0.6 from source code.
3. Compiled and installed LuaJIT from source code.
4. Compiled and installed openssl-1.1.1d from source code.
5. Compiled and installed snort-2.9.15.1 from source code
6. Configured snort per Snort 2.9.15.1 on CentOS 7 guide on https://www.snort.org/documents.
7. Compiled and installed https://github.com/ntop/PF_RING.git from source code per https://www.ntop.org/guides/pf_ring/thirdparty/snort-daq.html
8. Compiled and installed the PF_RING sample applications including pfcount.

This is where I came here for help because when I tried to use PF_RING with snort, I got the error message from my first post.

I then attempted to follow @cardigliano's suggestion per https://www.ntop.org/guides/pf_ring/get_started/packages_installation.html as follows:

1. I did not attempt a "yum install pfring" because I already compiled it in step 7 above.
2. I ran "pf_ringcfg --list-interfaces" and found the interface I wanted used the igb driver.
3. I then ran "pf_ringcfg --configure-driver igb --rss-queues 1", which did not gracefully terminate.  The error message in the yum history was "(ii)	warning:  %(post(igb-zc-5.3.5.18.2926-dkms.noarch) scriptlet failed, signal 1".

Further investigation showed that the script never got to the "configure-driver" part because it failed when it attempted "yum install pfring-dkms-7.6.0-2926.noarch.rpm igb-zc-5.3.5.18.2926-dkms.noarch.rpm".  It actually successfully installed pfring-dkms, but aborted during the igb-zc installation.  I have since undid both of these installations.

Hopefully this provides enough information for someone to help.  Thank you.

**Comment by cardigliano** [comment]
@dknobloc so it seems your problem now is that igb-zc installation is failing, the system you are running is RHEL Server Release 7.8 right? I will try to reproduce this on Centos 7 (we do not support RH)

**Comment by dknobloc** [comment]
Correct, the igb-zc installation is failing.  I am indeed running RHEL Server 7.8.

**Comment by cardigliano** [comment]
It seems to be working on Centos 7 (3.10.0-1127.el7.x86_64), are you using the dev repo?
Could you also provide "dkms status"?

**Comment by dknobloc** [comment]
Here are the repositories it is using for igb-zc, pfring-dkms, and their dependencies:

 igb-zc                                                ntop-noarch
 pfring-dkms                                      ntop-noarch  
dependencies:
 dkms                                                       epel
 elfutils-libelf-devel                            rhel-7-server-rpms  
 ndpi                                                           ntop 
 numactl                                            rhel-7-server-rpms 
 pfring                                                          ntop   

The other repos I have enabled are as follows:

rhel-7-server-extras-rpms
rhel-7-server-optional-rpms

I tried it again, this time installing all the dependencies first before attempting igb-zc.  All dependencies installed with no errors.  After I installed dkms, I ran "dkms status" and "sudo dkms status".  For both cases, the command returned no output at all.

I then attempted to install igb-zc with the same result.  It hung as it did before.  I have attached the "yum history info" record for this attempt.
[yum-history-info-46--2020-04-27.txt](https://github.com/ntop/PF_RING/files/4542705/yum-history-info-46--2020-04-27.txt)
